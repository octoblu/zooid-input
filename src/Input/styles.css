.root {
  display: flex;
  flex-direction: column;
  position: relative;
  font-size: 1em;

  box-sizing: border-box;
}

.root > *,
.root > *:before,
.root > *:after {
  box-sizing: inherit;
}


.input {
  border-radius: 2px;
  border: 1px solid #e6eaed;
  display: block;
  padding: 0.5em;
  width: 100%;
  margin-bottom: 0.25em;
  font-size: 1em;
  background: #fff;
  font-family: "citrixsans-regular", sans-serif;
  -webkit-appearance: none;
}

.input:focus {
  border: 1px solid #14568f;
  outline: none;
  background-color: #f8fdff;
  box-shadow: inset 0 1px 1px rgba(0, 0, 0, 0.075), 0 0 0 3px rgba(19, 133, 229, 0.1);
  transition: all 0.2s ease-in;
}

.input[disabled] {
  background-color: #f3f6f7;
  cursor: not-allowed;
}

.input[required]:focus {
  border: 1px solid #fec57b;
  background-color: #fff9f2;
}

.description {
  font-size: 0.8em;
  color: #6e798b;
}
