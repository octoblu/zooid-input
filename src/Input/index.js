import React, { PropTypes } from 'react'
import classNames from 'classnames'

import InputLabel from '../InputLabel/'
import styles from './styles.css'

const propTypes = {
  className: PropTypes.string,
  disabled: PropTypes.bool,
  description: PropTypes.string,
  label: PropTypes.string,
  name: PropTypes.string,
  placeholder: PropTypes.string,
  required: PropTypes.bool,
}

const defaultProps = {
  disabled: false,
  name: '',
  required: false,
}

class Input extends React.Component {
  constructor(props) {
    super(props)
  }

  render() {
    const {
      className,
      disabled,
      description,
      label,
      name,
      placeholder,
      required,
    } = this.props

    const classes = classNames(
      styles.root,
      { [`${styles['is-disabled']}`]: disabled },
      { [`${styles['is-required']}`]: required },
      className
    )

    return (
      <div className={classes}>
        <InputLabel
          name={name}
          required={required}
          description={description}
        >
          {label}
        </InputLabel>

        <input
          {...this.props}
          className={styles.input}
          disabled={disabled}
          name={name}
          placeholder={placeholder}
          required={required}
          type="text"
        />
        {description && <span className={styles.description}>{description}</span>}
      </div>
    )
  }
}

Input.propTypes    = propTypes
Input.defaultProps = defaultProps

export default Input
