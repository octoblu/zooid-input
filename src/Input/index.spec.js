import React from 'react'
import renderer from 'react-test-renderer'


import Input from './index'

describe('<Input />', () => {
  describe('when given no props', () => {
    it('should render', () => {
      const component = renderer.create(<Input />)
      const tree      = component.toJSON()
      expect(tree).toMatchSnapshot()
    })
  })

  describe('when given className prop', () => {
    it('should sanitize and merge props on the input component', () => {
      const component = renderer.create(
        <Input
          className="CustomClassName"
          description="Name your device"
          label="Device Name"
          name="deviceName"
          placeholder="Device name ish..."
        />
      )
      const tree = component.toJSON()

      expect(tree.props.className).toContain('CustomClassName')
      // expect(input).to.not.have.prop('className', 'CustomClassName')
      // expect(input).to.not.have.prop('description', 'Name your device')
      // expect(input).to.not.have.prop('label', 'Device Name')
      //
      // expect(input).to.have.prop('name', 'deviceName')
      // expect(input).to.have.prop('placeholder', 'Device name ish...')
    })
  })

})
