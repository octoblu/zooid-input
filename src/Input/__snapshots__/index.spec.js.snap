exports[`<Input /> when given no props should render 1`] = `
<div
  className="styles__formField___9sL2z root">
  <input
    className="input"
    disabled={false}
    name=""
    required={false}
    type="text" />
</div>
`;
